src/classes/application.js

```diff
@@ -160,6 +161,21 @@ module.exports = class Application extends Base {
     const app = express()
     _app.set(this, app)
 
+    log.trace('Ash server adding body parsing middleware')
+    const bodyParserOptions = config.bodyParser || {}
+    if (bodyParserOptions.json !== false) {
```

Comment:
This seems fragile to me, from the end-user perspective. I think I'd rather see more formal value checking. For instance, your example here actually makes an incorrect assumption about what happens when you omit an option.

```js
  ENV.bodyParser = {
    json: {type: 'json'},
    text: {type: 'text/html'}
    // raw: false,
    // urlencoded: false
  }
```

Results in the following, which does not match the implication in your example that omitting an option results in the option not being configured. On these grounds, I'd say this is a 10/10

```js
if (bodyParserOptions.urlencoded !== false) {
  app.use(bodyparser.urlencoded(Object.assign({extended: false}, bodyParserOptions.urlencoded)))
}

// turns into

if (undefined !== false) { // this if statement is true
  // this block gets run
  app.use(bodyparser.urlencoded(Object.assign({extended: false}, undefined)))
}
```


### Solution:

1. in the if statements, look for a truthy value instead of `false`
2. Optional: do at least some basic validation of the configuration value, and possibly throw an error if the config is malformed. This falls under the category of "validation analysis on app load" that we spoke about when I gave you the list of ideas I was thinking about. The result is that the app simply won't start if the configuration is incorrect.